pkg/workflow/manager.go

```diff
@@ -209,6 +219,17 @@ func (w *WorkflowManager) syncWorkflow(key string) error {
 		return err
 	}
 	workflow := *obj.(*api.Workflow)
+	// Set defaults for workflow
+	if _, ok := workflow.Labels[workflowUID]; ok == false {
+		newWorkflow, err := w.setLabels(&workflow)
```

Comment:
Why do we send a pointer here if the function returns a new workflow object? efficiency?